src/meta/processors/job/LeaderBalanceJobExecutor.cpp

```diff
@@ -0,0 +1,550 @@
+
+nebula::cpp2::ErrorCode LeaderBalanceJobExecutor::getAllSpaces(
+    std::vector<std::tuple<GraphSpaceID, int32_t, bool>>& spaces) {
+  // Get all spaces
+  folly::SharedMutex::ReadHolder rHolder(LockUtils::spaceLock());
+  const auto& prefix = MetaKeyUtils::spacePrefix();
+  std::unique_ptr<kvstore::KVIterator> iter;
+  auto retCode = kvstore_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &iter);
+  if (retCode != nebula::cpp2::ErrorCode::SUCCEEDED) {
+    LOG(ERROR) << "Get all spaces failed, error: " << apache::thrift::util::enumNameSafe(retCode);
+    return retCode;
+  }
+
+  while (iter->valid()) {
+    auto spaceId = MetaKeyUtils::spaceId(iter->key());
+    auto properties = MetaKeyUtils::parseSpace(iter->val());
+    bool zoned = !properties.get_zone_names().empty();
+    spaces.emplace_back(spaceId, *properties.replica_factor_ref(), zoned);
+    iter->next();
+  }
+  return nebula::cpp2::ErrorCode::SUCCEEDED;
+}
+
+ErrorOr<nebula::cpp2::ErrorCode, bool> LeaderBalanceJobExecutor::getHostParts(GraphSpaceID spaceId,
+                                                                              bool dependentOnZone,
+                                                                              HostParts& hostParts,
+                                                                              int32_t& totalParts) {
+  folly::SharedMutex::ReadHolder rHolder(LockUtils::spaceLock());
+  const auto& prefix = MetaKeyUtils::partPrefix(spaceId);
+  std::unique_ptr<kvstore::KVIterator> iter;
+  auto retCode = kvstore_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &iter);
+  if (retCode != nebula::cpp2::ErrorCode::SUCCEEDED) {
+    LOG(ERROR) << "Access kvstore failed, spaceId " << spaceId << " "
+               << apache::thrift::util::enumNameSafe(retCode);
+    return retCode;
+  }
+
+  while (iter->valid()) {
+    auto key = iter->key();
+    PartitionID partId;
+    memcpy(&partId, key.data() + prefix.size(), sizeof(PartitionID));
+    auto partHosts = MetaKeyUtils::parsePartVal(iter->val());
+    for (auto& ph : partHosts) {
+      hostParts[ph].emplace_back(partId);
+    }
+    totalParts++;
+    iter->next();
+  }
+
+  LOG(INFO) << "Host size: " << hostParts.size();
+  auto key = MetaKeyUtils::spaceKey(spaceId);
+  std::string value;
+  retCode = kvstore_->get(kDefaultSpaceId, kDefaultPartId, key, &value);
+  if (retCode != nebula::cpp2::ErrorCode::SUCCEEDED) {
+    LOG(ERROR) << "Access kvstore failed, spaceId " << spaceId
+               << apache::thrift::util::enumNameSafe(retCode);
+    return retCode;
+  }
+
+  auto properties = MetaKeyUtils::parseSpace(value);
+  if (totalParts != properties.get_partition_num()) {
+    LOG(ERROR) << "Partition number not equals " << totalParts << " : "
+               << properties.get_partition_num();
+    return false;
+  }
+
+  int32_t replica = properties.get_replica_factor();
+  LOG(INFO) << "Replica " << replica;
+  if (dependentOnZone && !properties.get_zone_names().empty()) {
```

Comment:
dependentOnZone and properties.get_zone_names().empty() are the same?

Reply:
This file is the historical code, it is only a move in this PR. The logic is not modified.